selftests/functional/test_variants.py

```diff
@@ -0,0 +1,75 @@
+import json
+import os
+import tempfile
+import shutil
+import unittest
+
+from avocado.utils import process
+
+
+basedir = os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', '..')
+basedir = os.path.abspath(basedir)
+
+AVOCADO = os.environ.get("UNITTEST_AVOCADO_CMD", "./scripts/avocado")
+
+
+class VariantsDumpLoadTests(unittest.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+        self.variants_file = os.path.join(self.tmpdir, 'variants.json')
+
+    def test_variants_dump(self):
+        content = ('[{"paths": ["/run/*"], '
+                   '"variant": [["/", []]], '
+                   '"variant_id": null}]')
+
+        cmd_line = ('%s variants --dump-variants %s' %
+                    (AVOCADO, self.variants_file))
+        process.run(cmd_line)
+        with open(self.variants_file, 'r') as file_obj:
+            self.assertEqual(file_obj.read(), content)
+
+    def test_run_dump(self):
+        content = ('[{"paths": ["/run/*"], '
+                   '"variant": [["/", []]], '
+                   '"variant_id": null}]')
+
+        cmd_line = ('%s run passtest.py --dump-variants %s '
+                    '--job-results-dir %s' %
+                    (AVOCADO, self.variants_file, self.tmpdir))
```

Comment:
you know me, I'm not a friend of duplicate tests, how about having `dump_and_run` test, check the dumped file and then run it? (it'd eliminate the one `avocado variants` execution but in case of failure it'd require manual check whether `run` works well in case of `dump` failure). Anyway this is really minor question...